src/editor/js/editor/index.js

```diff
@@ -0,0 +1,102 @@
+export class Editor {
+    static ROWS_COUNT_DEFAULT = 16;
+    static COLS_COUNT_DEFAULT = 16;
+    static COLOUR_DEFAULT = '#000000';
+    static CELL_WIDTH_DEFAULT = 20;
+    
+    canvas;
+    
+    #isDrawing = false;
+    #colour = Editor.COLOUR_DEFAULT;
+    #width = Editor.CELL_WIDTH_DEFAULT;
+    
+    constructor() {
+        this.build();
+    }
+    
+    /**
+     * Build cells
+     * @param rows {number} - count of rows
+     * @param cols {number} - count of cols
+     *
+     * @return void
+     */
+    build(
+        rows = Editor.ROWS_COUNT_DEFAULT,
+        cols = Editor.COLS_COUNT_DEFAULT
+    ) {
+        this.canvas = document.createElement('div');
+        this.canvas.classList.add('canvas');
+        
+        for (let r = 0; r < rows; r++) {
+            const row = document.createElement('div');
+            row.classList.add('canvas__row');
+            this.canvas.appendChild(row);
+            
+            for (let c = 0; c < cols; c++) {
+                const cell = document.createElement('div');
+                cell.classList.add('canvas__cell');
+                cell.style.width = `${this.#width}px`;
+                cell.style.height = `${this.#width}px`;
+                row.appendChild(cell);
+            }
+        }
+        
+        this.canvas.addEventListener('mousedown', this.#onPointerDown);
+        this.canvas.addEventListener('touchstart', this.#onPointerDown);
+        this.canvas.addEventListener('mousemove', this.#onPointerMove);
+        this.canvas.addEventListener('touchmove', this.#onPointerMove);
+        this.canvas.addEventListener('mouseup', this.#onPointerUp);
+        this.canvas.addEventListener('touchend', this.#onPointerUp);
+    }
+    
+    destroy() {
+        this.canvas.parentNode.removeChild(this.canvas);
+    
+        this.canvas.removeEventListener('mousedown', this.#onPointerDown);
+        this.canvas.removeEventListener('touchstart', this.#onPointerDown);
+        this.canvas.removeEventListener('mousemove', this.#onPointerMove);
+        this.canvas.removeEventListener('touchmove', this.#onPointerMove);
+        this.canvas.removeEventListener('mouseup', this.#onPointerUp);
+        this.canvas.removeEventListener('touchend', this.#onPointerUp);
+        
+        this.canvas = undefined;
+    }
+    
+    setup({
+      colour = this.#colour,
+      width = this.#width
+    }) {
+        this.#colour = colour;
+    
+        if (this.#width !== width) {
```

Comment:
Maybe add a comment of what we are doing here?
Also, maybe consider to refactor wifth stuff  to method like `setCellWidth`